knative/deployer.go

```diff
@@ -149,15 +149,25 @@ func probeFor(url string) *corev1.Probe {
 	}
 }
 
+func setHealthEndpoints(f fn.Function, c *corev1.Container) *corev1.Container {
+	if f.HealthEndpoints != nil {
+		if f.HealthEndpoints["liveness"] != "" {
+			c.LivenessProbe = probeFor(f.HealthEndpoints["liveness"])
+		}
+		if f.HealthEndpoints["readiness"] != "" {
+			c.ReadinessProbe = probeFor(f.HealthEndpoints["readiness"])
+		}
+	}
+	return c
+}
+
 func generateNewService(f fn.Function) (*servingv1.Service, error) {
 	containers := []corev1.Container{
 		{
 			Image: f.ImageWithDigest(),
 		},
 	}
-
-	containers[0].LivenessProbe = probeFor("/health/liveness")
-	containers[0].ReadinessProbe = probeFor("/health/readiness")
+	setHealthEndpoints(f, &containers[0])
```

Comment:
Perhaps not the time for this, but all this indexing into a slice is nerve-wracking.  It's one of those things begging for a future error.

This should probably be refactored to be a single container, because at no point in this function is it used in any way other than a single container, until line 204, at which point would be more appropriate to set it as the only member of slice defined inline anonymously.  The complexity starting to compound as now the `setHealthEndpoints` function needs to pass a reference, making this indexing now pass through a function boundary.

Perhaps better done during a dedicated refactor pass, but thought I'd mention it now, since the function's being touched, that's usually the best time to spruce things up.